### PR TITLE
[DS-1015] Loans, Insurance, Direct Payments

### PR DESCRIPTION
**Pull request by ejgullo**
- Displaying the three types in search results
- Showing Award pages for each type
- Changed Grant component into generic financial assistance component to allow all four to utilize it
- Added specific AwardAmounts component for Loans, as it does not contain the bar visualization
- Updating padding in tables for both search results and award transactions, as per designer request
- Removing Additional Information tab (and link) on non-contract award pages
- Removing Parent Award ID on non-contract pages

### CODE REVIEW

File: src/_scss/pages/award/table/cells/_genericHeader.scss [hunk 1]
```diff
@@ -22,7 +22,7 @@
             .header-icons {
                 display: table-cell;
                 vertical-align: middle;
-                padding-left: 10px;
+                padding-left: 5px;
```
Comment (ejgullo) [inline src/_scss/pages/award/table/cells/_genericHeader.scss]: 👍

File: src/js/components/award/AwardAmounts.jsx [hunk 2]
```diff
@@ -50,21 +51,41 @@ export default class AwardAmounts extends React.Component {
...
-            ceiling: this.formatFriendlyString(ceiling),
-            current: this.formatFriendlyString(current)
-        };
+        if (this.props.typeString === 'grant' || this.props.typeString === 'direct payment' ||
+        this.props.typeString === 'insurance') {
+            awardNarrative = (<p>This {this.props.typeString} was awarded to&nbsp;
+            <b className="recipient-name">{recipient}</b>
+            &nbsp;for <b>{current}</b>.</p>);
+        }
+        if (this.props.typeString === 'loan') {
```
Comment (ejgullo) [inline src/js/components/award/AwardAmounts.jsx]: 👍

File: src/js/components/award/details/DetailsTabBar.jsx [hunk 3]
```diff
@@ -8,39 +8,38 @@ import React from 'react';
...
-        }
-    ]
+    clickTab: React.PropTypes.func,
+    type: React.PropTypes.string
 };
 
 export default class DetailsTabBar extends React.Component {
     render() {
-        const tabs = this.props.tabs.map((tab) => (
+        const tabOptions = [
```
Comment (ejgullo) [inline src/js/components/award/details/DetailsTabBar.jsx]: 👍

File: src/js/dataMapping/search/tableSearchFields.js [hunk 4]
```diff
@@ -1,13 +1,11 @@
...
     columnWidths: {
-        award_id: 160,
-        recipient_name: 310,
-        period_of_performance_start_date: 150,
-        period_of_performance_current_end_date: 150,
-        total_obligation: 220,
-        type: 230,
-        awarding_agency_name: 280,
-        awarding_subtier_name: 280,
+        award_id: 120,
```
Comment (kevinli-work) [inline src/js/dataMapping/search/tableSearchFields.js]: Please restore the wider column widths. These are necessary to prevent the header cells from wrapping on fallback fonts.

File: src/js/dataMapping/search/tableSearchFields.js [hunk 5]
```diff
@@ -176,11 +151,9 @@ const tableSearchFields = {
             'recipient_name',
```
Comment (kevinli-work) [inline src/js/dataMapping/search/tableSearchFields.js]: Can you update the loan tab to use Geoff's columns:

* Award ID
* Recipient Name
* Issued Date
* Loan Value
* Subsidy Cost
* Funding Agency

If this is going to take too long, branch off this branch and do a separate PR with these changes.

Comment (kevinli-work) [inline src/js/dataMapping/search/tableSearchFields.js]: You will also need to add some logic to handle sorting by a column in one tab, then switching to a different tab where that column does not exist.

File: src/js/dataMapping/financialAssistance/financialAssistanceTransactionTable.js [hunk 6]
```diff
@@ -1,14 +1,14 @@
...
  * Created by Lizzie Dabbs 03/06/17
  */
 
 const tableSearchFields = {
     columnWidths: {
-        modification: 230,
-        action_date: 200,
-        amount: 200,
-        action_type: 380,
+        modification: 200,
```
Comment (kevinli-work) [inline src/js/dataMapping/financialAssistance/financialAssistanceTransactionTable.js]: See comment about needing wider columns

File: src/js/dataMapping/contracts/financialSystem.js [hunk 7]
```diff
@@ -5,10 +5,10 @@
 
 const tableFields = {
     columnWidths: {
-        submissionDate: 250,
-        tas: 250,
-        programActivity: 300,
-        objectClass: 300,
+        submissionDate: 160,
```
Comment (kevinli-work) [inline src/js/dataMapping/contracts/financialSystem.js]: See comment about needing wider columns.

File: src/js/dataMapping/contracts/transactionTable.js [hunk 8]
```diff
@@ -5,9 +5,9 @@
 
 const tableSearchFields = {
     columnWidths: {
-        modification: 230,
-        action_date: 200,
-        amount: 220,
+        modification: 200,
```
Comment (kevinli-work) [inline src/js/dataMapping/contracts/transactionTable.js]: See comment about needing wider columns.

File: src/js/dataMapping/financialAssistance/loanTransactionTable.js [hunk 9]
```diff
@@ -0,0 +1,57 @@
+/**
+ * loanTransactionTable.js
+ * Created by Emily Gullo 04/24/2017
+ */
+
+const tableSearchFields = {
+    columnWidths: {
```
Comment (kevinli-work) [inline src/js/dataMapping/financialAssistance/loanTransactionTable.js]: Please validate the column widths by temporarily disabling the custom font (fall back to default `sans-serif`) and testing on a variety of browsers to determine the smallest acceptable width for each. Add a bit of extra width for leeway.

File: src/js/components/award/details/DetailsTabBar.jsx [hunk 10]
```diff
@@ -8,39 +8,40 @@ import React from 'react';
...
-    ]
-};
+const tabOptions = [
+    {
+        label: 'Transaction History',
+        code: 'transaction',
+        disabled: false
+    },
+    {
+        label: 'Sub-Awards',
```
Comment (kevinli-work) [inline src/js/components/award/details/DetailsTabBar.jsx]: Why is subawards disabled? Subawards is implemented in dev, please make sure you are up-to-date and have reconciled the changes. Additionally, check to see which award types should have subawards and display as appropriate.

File: src/js/components/award/AwardAmounts.jsx [hunk 11]
```diff
@@ -50,21 +51,41 @@ export default class AwardAmounts extends React.Component {
...
-        const current = accounting.unformat(this.props.selectedAward.total_obligation);
+            MoneyFormatter.formatMoneyWithPrecision(
+                this.props.selectedAward.potential_total_value_of_award, 0);
+        const current = MoneyFormatter.formatMoneyWithPrecision(
+            this.props.selectedAward.total_obligation, 0);
 
         // calculate the percentage spent
         let percentage = 'N/A';
         if (ceiling && ceiling !== 0) {
             percentage = Math.floor((current / ceiling) * 1000) / 10;
```
Comment (kevinli-work) [inline src/js/components/award/AwardAmounts.jsx]: You can't do this any more because you've converted `current` and `ceiling` to currency-formatted strings. You need to perform the percentage calculating using real numbers.

This breaks the contract page (award 69626):
> This contract was awarded to Carahsoft Technology Corporation with a ceiling of  $2,856.  Of this amount, NaN% ($2,856) has been obligated.

File: src/js/dataMapping/search/tableSearchFields.js (hunk 4 above)
Comment (ejgullo) [inline src/js/dataMapping/search/tableSearchFields.js]: 👍

File: src/js/dataMapping/financialAssistance/financialAssistanceTransactionTable.js (hunk 6 above)
Comment (ejgullo) [inline src/js/dataMapping/financialAssistance/financialAssistanceTransactionTable.js]: 👍

File: src/js/dataMapping/contracts/financialSystem.js (hunk 7 above)
Comment (ejgullo) [inline src/js/dataMapping/contracts/financialSystem.js]: 👍

File: src/js/dataMapping/search/tableSearchFields.js (hunk 5 above)
Comment (ejgullo) [inline src/js/dataMapping/search/tableSearchFields.js]: Will branch off for this

File: src/js/dataMapping/contracts/transactionTable.js (hunk 8 above)
Comment (ejgullo) [inline src/js/dataMapping/contracts/transactionTable.js]: 👍

File: src/js/dataMapping/financialAssistance/loanTransactionTable.js (hunk 9 above)
Comment (ejgullo) [inline src/js/dataMapping/financialAssistance/loanTransactionTable.js]: 👍

File: src/js/components/award/details/DetailsTabBar.jsx (hunk 10 above)
Comment (ejgullo) [inline src/js/components/award/details/DetailsTabBar.jsx]: Mistake on a merge, fixing.

File: src/js/components/award/AwardAmounts.jsx (hunk 11 above)
Comment (ejgullo) [inline src/js/components/award/AwardAmounts.jsx]: 👍
